utils/swift_build_support/swift_build_support/targets.py

```diff
@@ -12,60 +12,115 @@
 import platform
 
 
-class StdlibDeploymentTarget(object):
+class Platform(object):
+    """
+    Abstract representation of a platform Swift can run on.
+    """
+
+    def __init__(self, name, archs, sdk_name=None, is_darwin=False):
+        """
+        Create a platform with the given name and list of architectures.
+        """
+        self.name = name
+        self.targets = [Target(self, arch) for arch in archs]
```

Comment:
The reason I did this is that I don't think there is an intrinsic requirement in the model that a platform contains targets which are organized by architecture, and I don't think clients should really assume that.

In practice, there can end up being other factors that need to impact the target definition (e.g., things like floating point behavior style, or platform ABI). So I wanted to take a step towards forcing the code to more accurately model all the things that may be eventually required.